app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -328,6 +338,46 @@ public void onClick(final SweetAlertDialog dialog) {
         }
     }
 
+    private void enableNFC() {
+        final NfcAdapter nfcAdapter = NfcAdapter.getDefaultAdapter(this);
+
+        if (nfcAdapter.isEnabled()) {
+            final Intent intent = new Intent(this, NFCActivity.class);
+            intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+            this.startActivity(intent);
+        } else {
+            final String description;
+            description = this.getString(R.string.popup_enable_nfc_exchange);
+
+            new SweetAlertDialog(this, SweetAlertDialog.NORMAL_TYPE)
+                    .setTitleText(this.getString(R.string.popup_enable_nfc_title))
+                    .setContentText(description)
+                    .setCancelText(this.getString(R.string.no))
+                    .setConfirmText(this.getString(R.string.yes))
+                    .setConfirmClickListener(new SweetAlertDialog.OnSweetClickListener() {
+                        @Override
+                        public void onClick(final SweetAlertDialog dialog) {
+                            dialog.dismiss();
+
+                            LOGGER.info("Requesting to enable NFC");
+                            new SweetAlertDialog(MainActivity.this, SweetAlertDialog.NORMAL_TYPE)
```

Comment:
I very much so dissaprove of this approach... what value does the second alert add? The first alert prompts the user with the message that they should enable NFC and the second alert prompts them with the request to do so...

Could you remove the second alert?